internal/machine/docker/server.go

```diff
@@ -529,9 +534,18 @@ func (s *Server) CreateServiceContainer(
 		containerName = fmt.Sprintf("%s-%s", spec.Name, suffix)
 	}
 
+	envVars := maps.Clone(spec.Container.Env)
```

Comment:
The tests are failing for a reason but error messages are not informative because the the `uncloudd` daemon panics when trying to set a key in `envVars` when it's `nil`.
```suggestion
	envVars := maps.Clone(spec.Container.Env)
	if envVars == nil {
		envVars = make(api.EnvVars)
	}
```